src/main/java/edu/kis/vh/nursery/collection/IntArrayStack.java

```diff
@@ -19,7 +19,7 @@ public boolean isEmpty() {
 
     @Override
     public boolean isFull() {
-        return total == NUMBERS_CAPACITY - 1;
+        return total == NUMBERS_CAPACITY + DEFAULT_EMPTY_STACK_VALUE;
```

Comment:
I do not see a reasonable relation between default value returned and isFull logic